modules/mediawiki/files/bin/deploy-mediawiki.py

```diff
@@ -167,138 +241,156 @@ def _construct_git_pull(repo: str, submodules: bool = False, branch: Optional[st
-        if args.l10n:  # setup l10n
-            if args.lang:
-                for language in str(args.lang).split(','):
-                    if not tag_is_valid(language):
-                        raise ValueError(f'{language} is not a valid language.')
-
-                lang = f'--lang={args.lang}'
-            else:
-                lang = ''
-
-            postinstall.append(f'sudo -u www-data php /srv/mediawiki/w/maintenance/mergeMessageFileList.php --quiet --wiki={envinfo["wikidbname"]} --output /srv/mediawiki/config/ExtensionMessageFiles.php')
-            rebuild.append(f'sudo -u www-data php /srv/mediawiki/w/maintenance/rebuildLocalisationCache.php {lang} --quiet --wiki={envinfo["wikidbname"]}')
-
-        for cmd in postinstall:  # cmds to run after rsync & install (like mergemessage)
-            exitcodes.append(run_command(cmd))
-        non_zero_code(exitcodes, nolog=args.nolog)
-        for cmd in rebuild:  # update ext list + l10n
-            exitcodes.append(run_command(cmd))
-        non_zero_code(exitcodes, nolog=args.nolog)
-
-        # see if we are online - exit code 3 if not
-        if args.port:
-            check_up(Debug=None, Host=envinfo['wikiurl'], verify=False, force=args.force, nolog=args.nolog, port=args.port)
-        else:
-            check_up(Debug=None, Host=envinfo['wikiurl'], verify=False, force=args.force, nolog=args.nolog)
-
-    # actually set remote lists
-    for option in options:
+    deploymentmap['loginfo'] = loginfo
+    deploymentmap['commands']['stage'] = _get_git_commands(args.world, args.pull, args.branch)
+    for option in options:  # configure rsync & custom data for repos
         if options[option]:
-            rsyncpaths.append(_get_deployed_path(option))
-    if args.files:
-        for file in str(args.files).split(','):
-            rsyncfiles.append(f'/srv/mediawiki/{file}')
-    if args.folders:
-        for folder in str(args.folders).split(','):
-            rsyncpaths.append(f'/srv/mediawiki/{folder}/')
-    if args.extensionlist:
-        rsyncfiles.append('/srv/mediawiki/cache/extension-list.json')
-    if args.l10n:
-        rsyncpaths.append('/srv/mediawiki/cache/l10n/')
-
-    for path in rsyncpaths:
-        exitcodes.append(remote_sync_file(time=args.ignoretime, serverlist=servers, path=path, force=args.force, envinfo=envinfo, nolog=args.nolog))
-    for file in rsyncfiles:
-        exitcodes.append(remote_sync_file(time=args.ignoretime, serverlist=servers, path=file, recursive=False, force=args.force, envinfo=envinfo, nolog=args.nolog))
-
-    fintext = f'finished deploy of "{str(loginfo)}" to {synced}'
+            if option == 'world':  # install steps for w
+                deploymentmap['commands']['rebuild'].append(WikiCommand('MW_INSTALL_PATH=/srv/mediawiki-staging/w /srv/mediawiki-staging/w/extensions/MirahezeMagic/maintenance/rebuildVersionCache.php --save-gitinfo --conf=/srv/mediawiki-staging/config/LocalSettings.php', envinfo['wikidbname']))
```

Comment:
Would this not be `sudo -u www-data php MW_INSTALL_PATH=/srv/mediawiki-staging/w ...` still? Whereas it should be `sudo -u www-data MW_INSTALL_PATH=/srv/mediawiki-staging/w php ...` I think? (The env before php)

Reply:
Yes, that's how wiki command will output it. Can you test to see if that breaks stuff? I can add support for injecting the env variable if not.